### PR TITLE
chore: AR-2542 publish storybooks to github page

### PR DESCRIPTION
**Pull request by aziz-zen-ab**
This pull request add `publish-storybook.yml` to publish our storybooks to the github page.

### CODE REVIEW

File: .github/workflows/publish-storybook.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        with:
+          persist-credentials: false
+      - name: Install and Build
+        run: | # Install npm packages and build the Storybook files
+          npm install --legacy-peer-deps
+          npm run build-storybook
+      - name: Deploy
+        uses: JamesIves/github-pages-deploy-action@3.6.2
+        with:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (rayhan-ab) [inline .github/workflows/publish-storybook.yml:21]: change this to secrets.GITHUB_TOKEN

**Comment by argaditya-ab** [comment]
I am curious, publishing storybook to github page means it will be accessible to everyone right? is this intended?

**Comment by rayhan-ab** [comment]
based on [this discussion](https://accelbyte.atlassian.net/wiki/spaces/PJ/pages/252379167?focusedCommentId=2433351813#comment-2433351813), yes

File: .github/workflows/publish-storybook.yml (hunk 1 above)
Comment (aziz-zen-ab) [reply .github/workflows/publish-storybook.yml:21]: Okay

**Comment by aziz-zen-ab** [comment]
@argaditya-ab Yes, do you have any concern about it?

**Pull request merged**